craft_providers/bases/centos.py

```diff
@@ -0,0 +1,950 @@
+        try:
+            # Replace encoding errors if it somehow occurs with utf-8. This
+            # doesn't need to be perfect for checking compatibility.
+            proc = executor.execute_run(
+                command=["cat", "/etc/os-release"],
+                capture_output=True,
+                check=True,
+                text=True,
+                encoding="utf-8",
+                errors="replace",
+            )
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to read /etc/os-release.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+        return parse_os_release(proc.stdout)
+
+    def _ensure_os_compatible(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Ensure OS is compatible with Base.
+
+        :raises BaseCompatibilityError: if instance is incompatible.
+        :raises BaseConfigurationError: on other unexpected error.
+        """
+        os_release = self._get_os_release(executor=executor, deadline=deadline)
+
+        os_id = os_release.get("ID")
+        if os_id not in ("centos", "rhel"):
+            raise BaseCompatibilityError(
+                reason=f"Expected OS 'centos', found {os_id!r}"
+            )
+
+        compat_version_id = self.alias.value[1]
+        version_id = os_release.get("VERSION_ID")
+
+        if version_id != compat_version_id:
+            raise BaseCompatibilityError(
+                reason=(
+                    f"Expected OS version {compat_version_id!r},"
+                    f" found {version_id!r}"
+                )
+            )
+
+    def get_command_environment(
+        self,
+    ) -> Dict[str, Optional[str]]:
+        """Get command environment to use when executing commands.
+
+        :returns: Dictionary of environment, allowing None as a value to
+                  indicate that a value should be unset.
+        """
+        return self.environment.copy()
+
+    def setup(
+        self,
+        *,
+        executor: Executor,
+        retry_wait: float = 0.25,
+        timeout: Optional[float] = None,
+    ) -> None:
+        """Prepare base instance for use by the application.
+
+        Wait for environment to become ready and configure it.  At completion of
+        setup, the executor environment should have networking up and have all
+        of the installed dependencies required for subsequent use by the
+        application.
+
+        The state of the setup is tracked in the instance config. If the setup is
+        interrupted, the state can be checked so that a partially setup instance
+        can be discarded.
+
+        Setup may be called more than once in a given instance to refresh/update
+        the environment.
+
+        If timeout is specified, abort operation if time has been exceeded.
+
+        Guarantees provided by this setup:
```

Comment:
Can you add anything noteworthy done during setup? Perhaps yum or the python setup.